src/parser/expression.js

```diff
@@ -390,8 +390,16 @@ pp.parseSubscripts = function (base, startPos, startLoc, noCalls) {
     } else if (this.hasPlugin("lightscript") && !noCalls && this.eat(tt.tilde)) {
       const node = this.startNodeAt(startPos, startLoc);
       node.left = base;
-      // allow Identifier or MemberExpression, but not calls
-      node.right = this.parseSubscripts(this.parseIdentifier(), this.state.start, this.state.startLoc, true);
+      let right;
+      if (this.state.value === 'this') {
```

Comment:
I think you probably want to check for `this.state.type == tt._this` instead of looking at `state.value`